go/exorelay/main.go

```diff
@@ -19,10 +19,16 @@ type Config struct {
 	Role string
 }
 
+// MessageHandler is the function signature for functions used for handling messages
+type MessageHandler func(structs.MessagePayload, func(string, structs.MessagePayload))
```

Comment:
Can you give a code example of the interface you are thinking? I've seen a couple use of channels but don't know how we would build an API with them